telemetry/peer_count.go

```diff
@@ -2,66 +2,84 @@ package telemetry
 
 import (
 	"database/sql"
+	"encoding/json"
+	"fmt"
 	"time"
+
+	"github.com/status-im/dev-telemetry/pkg/types"
 )
 
 type PeerCount struct {
-	ID            int    `json:"id"`
-	CreatedAt     int64  `json:"createdAt"`
-	Timestamp     int64  `json:"timestamp"`
-	NodeName      string `json:"nodeName"`
-	NodeKeyUid    string `json:"nodeKeyUid"`
-	PeerID        string `json:"peerId"`
-	PeerCount     int    `json:"peerCount"`
-	StatusVersion string `json:"statusVersion"`
+	data types.PeerCount
```

Comment:
Instead of creating a `data` attribute, wdyt about using
```
type PeerCount struct {
   types.PeerCount
```
And the same for `PeerConnFailure`: 
```
type PeerConnFailure struct {
   types.PeerCount
```


I might be wrong but I think this would let you access the attributes directly, i.e:
`myPeerConnFailure.Timestamp`, instead of `myPeerConnFailure.data.Timestamp`

Reply:
Sadly, not the case, I'd have to implement the getters for the values - calling methods on composed types works, but to access the attributes, you need to do something like

```
type PeerCount struct {
	types.PeerCount
}

func (r *PeerCount) process(db *sql.DB, errs *MetricErrors, data *types.TelemetryRequest) error {
...
	lastInsertId := 0
	err = stmt.QueryRow(
		r.PeerCount.Timestamp,
...
```

So it is not really solving much:( The only benefit is that I can use `json.Unmarshal(*data.TelemetryData, &r);` instead of `&r.data`, but that seems to be the only benefit

Reply:
:(